### PR TITLE
DBTP-508 - Remove conduit resources when container shuts down

### PR DESCRIPTION
**Pull request by JohnStainsby**
A CloudFormation stack and a bunch of resources are created when a conduit session starts. This PR makes the conduit container delete these before it shuts down.

- Add policy to default task role to allow deletion
- Install AWS CLI in conduit image
- Trigger stack to delete before shutdown

### CODE REVIEW

File: dbt_copilot_helper/commands/conduit.py [hunk 1]
```diff
@@ -152,6 +153,33 @@ def connect_to_addon_client_task(app: str, env: str, cluster_arn: str, addon_nam
...
+                        "Statement": [
+                            {
+                                "Action": ["cloudformation:DeleteStack"],
+                                "Effect": "Allow",
+                                "Resource": f"arn:aws:cloudformation:*:*:stack/{conduit_stack_name}/*",
+                            }
+                        ],
+                    }
+                ),
+            )
```
Comment (acodeninja) [inline dbt_copilot_helper/commands/conduit.py:180]: Just for consistency

```suggestion
            boto3.client("iam").put_role_policy(
                RoleName=task_role_name,
                PolicyName="DeleteCloudFormationStack",
                PolicyDocument=json.dumps(
                    {
                        "Version": "2012-10-17",
                        "Statement": [
                            {
                                "Action": ["cloudformation:DeleteStack"],
                                "Effect": "Allow",
                                "Resource": f"arn:aws:cloudformation:*:*:stack/{conduit_stack_name}/*",
                            },
                        ],
                    },
                ),
            )
```

File: images/conduit/opensearch/Dockerfile [hunk 2]
```diff
@@ -10,4 +10,10 @@ COPY entrypoint.sh /entrypoint.sh
 ADD https://artifacts.opensearch.org/opensearch-clients/opensearch-cli/opensearch-cli-${OPENSEARCH_CLI_VERSION}-linux-arm64.zip /opensearch-cli.zip
 RUN cd /root && unzip /opensearch-cli.zip
 
+# AWS CLI
+RUN curl "https://awscli.amazonaws.com/awscli-exe-linux-aarch64.zip" -o "awscliv2.zip" && \
```
Comment (acodeninja) [inline images/conduit/opensearch/Dockerfile:14]: You can save yourself from also installing `curl` by using the `ADD` directive:

```Dockerfile
ADD https://awscli.amazonaws.com/awscli-exe-linux-aarch64.zip ./awscliv2.zip
```

**Review by acodeninja — COMMENTED** [review]
Few tidy up things and a suggestion on testing and use of Docker's `ADD` directive.

File: images/conduit/opensearch/Dockerfile (hunk 2 above)
Comment (JohnStainsby) [reply images/conduit/opensearch/Dockerfile:14]: Good to know. It's also right there on the line above, don't know how I didn't see it!

Comment (acodeninja) [reply images/conduit/opensearch/Dockerfile:14]: The other advantage with this is that if the cache invalidation headers for the URL haven't changed, docker will not re-run the add command.

**Comment by acodeninja** [comment]
When you update to main, remember to bump the version number, just released another version.

File: images/conduit/opensearch/Dockerfile (hunk 2 above)
Comment (JohnStainsby) [reply images/conduit/opensearch/Dockerfile]: I've swapped to using the ADD command but curl is required later so will still need to install.

File: images/conduit/opensearch/Dockerfile [hunk 3]
```diff
@@ -10,4 +10,10 @@ COPY entrypoint.sh /entrypoint.sh
...
 RUN cd /root && unzip /opensearch-cli.zip
 
-ENTRYPOINT ["bash", "/entrypoint.sh"]
+# AWS CLI
+ADD https://awscli.amazonaws.com/awscli-exe-linux-aarch64.zip /awscliv2.zip
+RUN unzip awscliv2.zip && \
+    ./aws/install && \
+    rm -rf aws && rm awscliv2.zip
+
+ENTRYPOINT ["bash", "/entrypoint.sh"]
```
Comment (acodeninja) [inline images/conduit/opensearch/Dockerfile:19]: No new line

**Review by acodeninja — APPROVED** [review]
One small change, otherwise great work!

**Review by WillGibson — APPROVED** [review]
Nice

**Pull request merged**